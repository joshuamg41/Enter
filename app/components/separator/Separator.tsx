import React, { FunctionComponent } from 'react';
import { View, ViewProps } from "react-native";
import METRICS from '../../themes/Metrics';
import { horizontalScale, verticalScale } from '../../utils/StyleHelpers';

const Separator: FunctionComponent<propTypes> = props => {
  return (
    <View
      style={{ height: verticalScale(props.height), backgroundColor: props.color, width: horizontalScale(props.width) }}
    />
  )
};

interface propTypes extends ViewProps {
  height?: number;
  color?: string;
  width?: number;
}

const defaultProps = {
  height: METRICS.xLarge20,
  color: 'transparent',
  width: METRICS.large15,
}

Separator.defaultProps = defaultProps;

export default React.memo(Separator);