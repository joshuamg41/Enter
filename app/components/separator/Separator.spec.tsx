import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Separator from './Separator';

describe('Separator', () => {
  it('MatchSnapshot', () => {
    const tree = renderer
      .create(<Separator />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
})