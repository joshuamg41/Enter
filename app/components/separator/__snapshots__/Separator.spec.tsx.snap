// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Separator MatchSnapshot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "transparent",
      "height": 39,
      "width": 32,
    }
  }
/>
`;
